packages/tables/src/elements/Cell.tsx

```diff
@@ -7,18 +7,28 @@
 
 import React from 'react';
 import PropTypes from 'prop-types';
-import { StyledCell } from '../styled';
+import { StyledCell, StyledHiddenCell } from '../styled';
 import { useTableContext } from '../utils/useTableContext';
 import { ICellProps } from '../types';
 
 /**
  * @extends TdHTMLAttributes<HTMLTableCellElement>
  */
-export const Cell = React.forwardRef<HTMLTableCellElement, ICellProps>((props, ref) => {
-  const { size } = useTableContext();
+export const Cell = React.forwardRef<HTMLTableCellElement, ICellProps>(
+  ({ hidden, ...props }, ref) => {
+    const { size } = useTableContext();
 
-  return <StyledCell ref={ref} size={size} {...props} />;
-});
+    return (
+      <StyledCell ref={ref} size={size} {...props}>
+        {hidden && props.children ? (
```

Comment:
#nitpick, is the `&& props.children` necessary here?

Reply:
If a consumer decides to use an `aria-label` on the cell instead of children, this prevents an empty `div`.